app/components/sidebars/main/channels_list/channel_item/__snapshots__/channel_item.test.js.snap

```diff
@@ -818,3 +808,821 @@ ShallowWrapper {
+                    "isInfo": false,
+                    "isUnread": true,
+                    "membersCount": 1,
+                    "size": 16,
+                    "status": "online",
+                    "theme": Object {
+                      "sidebarText": "#aaa",
+                      "sidebarTextActiveBorder": "#aaa",
+                      "sidebarTextActiveColor": "#aaa",
+                      "sidebarTextHoverBg": "#aaa",
+                    },
+                    "type": "D",
+                  },
+                  "ref": null,
+                  "rendered": null,
+                  "type": [Function],
+                },
+                Object {
+                  "instance": null,
+                  "key": undefined,
+                  "nodeType": "class",
+                  "props": Object {
+                    "accessible": true,
+                    "allowFontScaling": true,
+                    "children": undefined,
+                    "ellipsizeMode": "tail",
+                    "numberOfLines": 1,
+                    "style": Array [
+                      Object {
+                        "color": "rgba(170,170,170,0.4)",
+                        "flex": 1,
+                        "fontSize": 14,
+                        "fontWeight": "600",
+                        "height": "100%",
+                        "lineHeight": 44,
+                        "paddingRight": 40,
+                        "textAlignVertical": "center",
+                      },
+                      Object {
+                        "color": undefined,
+                      },
+                    ],
+                  },
+                  "ref": null,
+                  "rendered": null,
+                  "type": [Function],
+                },
+                undefined,
+              ],
+              "type": [Function],
+            },
+          ],
+          "type": [Function],
+        },
+        "type": [Function],
+      },
+      "type": [Function],
+    },
+  ],
+  Symbol(enzyme.__options__): Object {
+    "adapter": ReactSixteenAdapter {
+      "options": Object {
+        "enableComponentDidUpdateOnSetState": true,
+      },
+    },
+    "context": Object {
+      "intl": Object {
+        "formatMessage": [MockFunction] {
+          "calls": Array [
+            Array [
+              Object {
+                "defaultMessage": "{displayName} (you)",
+                "id": "channel_header.directchannel.you",
+              },
+              Object {
+                "displayname": "display_name",
+              },
+            ],
+            Array [
+              Object {
+                "defaultMessage": "{displayname} - Deactivated",
+                "id": "more_direct_channels.directchannel.deactivated",
+              },
+              Object {
+                "displayname": undefined,
+              },
+            ],
+          ],
+        },
+      },
+    },
+  },
+}
+`;
```

Comment:
Just a note, I'm not sure how but I think we need to improve our snapshot tests on mobile RN.  It's adding too much LOCs and I'm not sure if it will save us for future iteration of this component.

Reply:
I don't about this either but it is a start. When we do face the problem of future iterations we might be able to come up with better solutions around i think. 

My suggestion is to have a lot of functional components if we are concerned about future iterations 

1. Connected components(Connected to redux)
2. Container components(Connected to redux and has state in it.)
3. Functional components(Just pure functions)

Last components in the hierarchy of any react components should probably always be a functional component.

One downside of that I faced previously with this approach is us ending up with a lot of components and a whole lot of files so searching for things becomes harder. All just opinions and up for discussion